### PR TITLE
Feature: Passthru metadata

### PR DESCRIPTION
**Pull request by Ognian**
**Please Read the [CONTRIBUTING Guidelines](https://github.com/babel/babel-loader/blob/master/CONTRIBUTING.md)**
*In particular the portion on Commit Message Formatting*

**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the following...

* Impact:
* Migration path for existing applications:
* Github Issue(s) this is regarding:

**Other information**:

### CODE REVIEW

**Comment by Ognian** [comment]
@danez I think that this pr can now be considered to be merged. Please answer in a timely manner, since now I'm up to date again and can quickly implement some changes if desired. You should close #329 too (I can't do this since I'm not the owner)
Thanks
Ognian

File: src/index.js [hunk 1]
```diff
@@ -134,10 +146,16 @@ module.exports = function(source, inputSourceMap) {
       transform: transpile,
     }, function(err, result) {
       if (err) { return callback(err); }
+      metadataSubscribers.map(function (s) {
```
Comment (danez) [inline src/index.js]: No nead for` .map()`. Can use `.forEach()`

File: src/index.js [hunk 2]
```diff
@@ -134,10 +146,16 @@ module.exports = function(source, inputSourceMap) {
...
+        passMetadata(s, context, result.metadata);
+      });
       return callback(null, result.code, result.map);
     });
   }
 
   const result = transpile(source, options);
+  metadataSubscribers.map(function (s) {
```
Comment (danez) [inline src/index.js]: Same here

File: .gitignore [hunk 3]
```diff
@@ -4,3 +4,4 @@ lib
 node_modules
 npm-debug.log
 test/output
+.idea
```
Comment (danez) [inline .gitignore]: Can we please revert this. You should ignore this folder globally on your machine.
https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: package.json [hunk 4]
```diff
@@ -32,6 +32,10 @@
     "eslint-plugin-flowtype": "^2.25.0",
     "nyc": "^10.0.0",
     "rimraf": "^2.4.3",
+    "react": "^15.1.0",
+    "react-intl":"^2.1.2",
+    "babel-plugin-react-intl": "2.1.3",
```
Comment (danez) [inline package.json]: Can we do `^2.1.3` here so we test agains the latest version 2.3.0

**Comment by mbrevda** [comment]
Waho 🎉 ! Thanks so much @Ognian! Are there docs/examples anywhere for how to access the metadata now?

**Pull request merged**